src/crawlee/abstract_http_crawler/_abstract_http_crawler.py

```diff
@@ -0,0 +1,204 @@
+from abc import ABC
+from typing import TYPE_CHECKING, Any, Generic
+
+from pydantic import ValidationError
+from typing_extensions import NotRequired, TypeVar
+
+from crawlee import EnqueueStrategy
+from crawlee._request import BaseRequestData
+from crawlee._utils.docs import docs_group
+from crawlee._utils.urls import convert_to_absolute_url, is_url_absolute
+from crawlee.abstract_http_crawler._http_crawling_context import (
+    HttpCrawlingContext,
+    ParsedHttpCrawlingContext,
+    TParseResult,
+)
+from crawlee.basic_crawler import BasicCrawler, BasicCrawlerOptions, ContextPipeline
+from crawlee.errors import SessionError
+from crawlee.http_clients import HttpxHttpClient
+
+if TYPE_CHECKING:
+    from collections.abc import AsyncGenerator, Iterable
+
+    from typing_extensions import Unpack
+
+    from crawlee._types import BasicCrawlingContext, EnqueueLinksFunction, EnqueueLinksKwargs
+    from crawlee.abstract_http_crawler._abstract_http_parser import AbstractHttpParser
+
+TCrawlingContext = TypeVar('TCrawlingContext', bound=ParsedHttpCrawlingContext)
+
+
+@docs_group('Data structures')
+class HttpCrawlerOptions(Generic[TCrawlingContext], BasicCrawlerOptions[TCrawlingContext]):
+    additional_http_error_status_codes: NotRequired[Iterable[int]]
+    """Additional HTTP status codes to treat as errors, triggering automatic retries when encountered."""
+
+    ignore_http_error_status_codes: NotRequired[Iterable[int]]
+    """HTTP status codes typically considered errors but to be treated as successful responses."""
+
+
+@docs_group('Abstract classes')
+class AbstractHttpCrawler(Generic[TCrawlingContext, TParseResult], BasicCrawler[TCrawlingContext], ABC):
+    """A web crawler for performing HTTP requests.
+
+    The `AbstractHttpCrawler` builds on top of the `BasicCrawler`, which means it inherits all of its features. On top
+    of that it implements the HTTP communication using the HTTP clients. The class allows integration with
+    any HTTP client that implements the `BaseHttpClient` interface. The HTTP client is provided to the crawler
+    as an input parameter to the constructor.
+    AbstractHttpCrawler is generic class and is expected to be used together with specific parser that will be used to
+    parse http response and type of expected TCrawlingContext which is available to the user function.
+    See prepared specific version of it: BeautifulSoupCrawler, ParselCrawler or HttpCrawler for example.
+
+    The HTTP client-based crawlers are ideal for websites that do not require JavaScript execution. However,
+    if you need to execute client-side JavaScript, consider using a browser-based crawler like the `PlaywrightCrawler`.
+    """
+
+    def __init__(
+        self,
+        *,
+        parser: AbstractHttpParser[TParseResult],
+        additional_http_error_status_codes: Iterable[int] = (),
+        ignore_http_error_status_codes: Iterable[int] = (),
+        **kwargs: Unpack[BasicCrawlerOptions[TCrawlingContext]],
```

Comment:
Can't we use `HttpCrawlerOptions` here?

```suggestion
        **kwargs: Unpack[HttpCrawlerOptions[TCrawlingContext]],
```

Reply:
This is actually strange. We pass those arguments (even before change in this PR) to BasicCrawler. But BasicCrawler does not use them. So I would either make BasicCrawler use them and then I can delete HttpCrawlerOptions and put those extra args into BasicCrawlerOptions or I should not pass them to BasicCrawler